be/src/olap/reader.cpp

```diff
@@ -532,33 +521,31 @@ OLAPStatus Reader::_capture_rs_readers(const ReaderParams& read_params) {
         }
     }
 
-    _reader_context.reader_type = read_params.reader_type;
-    _reader_context.tablet_schema = &_tablet->tablet_schema();
-    _reader_context.need_ordered_result = need_ordered_result;
-    _reader_context.return_columns = &_return_columns;
-    _reader_context.seek_columns = &_seek_columns;
-    _reader_context.load_bf_columns = &_load_bf_columns;
-    _reader_context.conditions = &_conditions;
-    _reader_context.predicates = &_col_predicates;
-    _reader_context.lower_bound_keys = &_keys_param.start_keys;
-    _reader_context.is_lower_keys_included = &_is_lower_keys_included;
-    _reader_context.upper_bound_keys = &_keys_param.end_keys;
-    _reader_context.is_upper_keys_included = &_is_upper_keys_included;
-    _reader_context.delete_handler = &_delete_handler;
-    _reader_context.stats = &_stats;
-    _reader_context.runtime_state = read_params.runtime_state;
-    _reader_context.use_page_cache = read_params.use_page_cache;
+    RowsetReaderContext reader_context;
+    reader_context.reader_type = read_params.reader_type;
+    reader_context.tablet_schema = &_tablet->tablet_schema();
+    reader_context.need_ordered_result = need_ordered_result;
+    reader_context.return_columns = &_return_columns;
+    reader_context.seek_columns = &_seek_columns;
+    reader_context.load_bf_columns = &_load_bf_columns;
+    reader_context.conditions = &_conditions;
+    reader_context.predicates = &_col_predicates;
+    reader_context.lower_bound_keys = &_keys_param.start_keys;
+    reader_context.is_lower_keys_included = &is_lower_keys_included;
```

Comment:
Is it safe to refer a local variable `is_lower_keys_included`?
Will the local `is_lower_keys_included` be deconstructed after this method?

Reply:
Oh, it's not safe, I've fixed it now.